source/math/FloatVectorList.ooc

```diff
@@ -381,4 +381,85 @@ FloatVectorList: class extends VectorList<Float> {
 			result add(this[i] clamp(floor, ceiling))
 		result
 	}
+	findMaxIndex: func -> Int {
+		index := 0
+		for (i in 1 .. this count)
+			if (this[index] < this[i])
+				index = i
+		index
+	}
+	crossCorrelation: func (other: This, vectorLength: Int) -> Float {
+		upperPart := 0.0f
+		lowerPart1 := 0.0f
+		lowerPart2 := 0.0f
+		thisMean := this mean
+		otherMean := other mean
+		thisPointer := (this pointer as Float*)
+		otherPointer := (other pointer as Float*)
+		for (i in 0 .. vectorLength) {
+			upperPart += ((thisPointer[i] - thisMean) * (otherPointer[i] - otherMean))
+			lowerPart1 += ((thisPointer[i] - thisMean) * (thisPointer[i] - thisMean))
+			lowerPart2 += ((otherPointer[i] - otherMean) * (otherPointer[i] - otherMean))
+		}
+		upperPart / sqrt(lowerPart1 * lowerPart2)
+	}
+	shift: func (offset: Int, filledValue := 0.0f) -> This {
```

Comment:
I started with memcpy, yet I must have done something wrong. Then I made it like this so as to save time. Because optimization can be done after the function is working.